app/src/main/java/org/mifos/mobilebanking/presenters/LoginPresenter.java

```
@@ -175,25 +175,27 @@ public void onNext(Page<Client> clientPage) {
 
 
     private boolean isCredentialsValid(final String username, final String password) {
-
+        boolean credentialValid = true;
         final Resources resources = context.getResources();
         final String correctUsername = username.replaceFirst("\\s++$", "").trim();
         if (username == null || username.matches("\\s*") || username.isEmpty()) {
             getMvpView().showUsernameError(context.getString(R.string.error_validation_blank,
                     context.getString(R.string.username)));
-            return false;
+            credentialValid = false;
         } else if (username.length() < 5) {
             getMvpView().showUsernameError(context.getString(R.string.error_validation_minimum_chars
                     , resources.getString(R.string.username), resources.getInteger(R.integer.
                             username_minimum_length)));
-            return false;
+            credentialValid = false;
         } else if (correctUsername.contains(" ")) {
             getMvpView().showUsernameError(context.getString(
                     R.string.error_validation_cannot_contain_spaces,
                     resources.getString(R.string.username),
                     context.getString(R.string.not_contain_username)));
-            return false;
-        } else if (password == null || password.isEmpty()) {
+            credentialValid = false;
+        }
```

@m-sameer Add else and hide errors there. See #907 for reference.

---

@m-sameer Remove [this](https://github.com/openMF/mifos-mobile/blob/development/app/src/main/java/org/mifos/mobilebanking/ui/activities/LoginActivity.java#L122)
and add else there remove the error.